src/nvim/message.c

```diff
@@ -160,6 +164,16 @@ static bool msg_ext_history = false;  ///< message was added to history
 
 static int msg_grid_pos_at_flush = 0;
 
+static MsgID msg_id_next = 1;           ///< message id to be allocated to next message
+
+// code inside doesn't emit ext-message
+#define NO_EXT_MSG(code) \
+  do { \
+    msg_ext_skip = true; \
+    code; \
+    msg_ext_skip = false; \
+  } while (0)
```

Comment:
this looks very sus. why is this necessary for progress messages but not other messages?

if we drop `draw_progress_message` and only support "extui", can we get rid of this? 

even without "extui", couldn't we show progress message by handling a `Progress` event in Lua ?

Reply:
This is mainly because of this

> - Progress message get's displayed in TUI/cmdline in the following format
`{title}: {msg}..{percent}%`

In tui I'm showing the msg in special format to include title and percent. In case of ext-ui events I'm sending those segments seperately in msg_show event. Basically whatever we write on tui msg_area it get's forwarded to ext-ui as msg. That's is handled quite more lower-level in code. Passing around a flag as arg seemed wayy more messy then this to me.

This basically just adds a file-local global flag that enables and disables that forwarding the macro just wrap the code with toggle for the flag.

Basically if we get rid of it we'll only show the msg without title and percent. Extui will still get all the info and it can show as they like